File: Python/Signaux/7.4.py

n = int(input("Nombre d'éléments : "))
tab = [int(input(f"tab[{i}] = ")) for i in range(n)]

while True:
    i = int(input("Index à afficher : "))
    if 0 <= i < n:
        print(f"tab[{i}] = {tab[i]}")
        break
    else:
        print("Erreur : index invalide !")

File: C/Signaux/7.4.c
#include <stdio.h>
#include <stdlib.h>

int main() 
{
    int n;
    printf("Nombre d'éléments : ");
    scanf("%d", &n);

    int *tab = malloc(n * sizeof(int));
    for(int i=0;i<n;i++) {
        printf("tab[%d] = ", i);
        scanf("%d", &tab[i]);
    }

    int i;
    while(1) {
        printf("Index à afficher : ");
        scanf("%d", &i);
        if(i>=0 && i<n) {
            printf("tab[%d] = %d\n", i, tab[i]);
            break;
        } else {
            printf("Erreur : index invalide !\n");
        }
    }

    free(tab);
    return 0;
}